erts/emulator/beam/erl_bif_persistent.c

```diff
@@ -953,19 +960,25 @@ cleanup_trap_data(Binary *bp)
 }
 
 static Uint
-lookup(HashTable* hash_table, Eterm key)
+lookup(HashTable* hash_table, Eterm key, Eterm *bucket)
 {
     Uint mask = hash_table->mask;
     Uint32 idx = make_internal_hash(key, 0);
     Eterm term;
 
     while (1) {
         term = get_bucket(hash_table, idx & mask);
-        if (is_nil(term) || EQ(key, (tuple_val(term))[1]))
-            break;
+
+        if (is_nil(term) || EQ(key, (tuple_val(term))[1])) {
+            if (bucket) {
+                *bucket = term;
+            }
```

Comment:
`bucket` argument is never NULL to `lookup()` what I can see

Reply:
Thanks, fixed. :)